compiler/AST/iterator.cpp

```diff
@@ -645,8 +696,7 @@ buildHasMore(IteratorInfo* ii, BlockStmt* singleLoop) {
     // In copied expressions, replace _ic with hasMore->_this .
     map.put(ic, ii->hasMore->_this);
 
-    // NOAKES 2014/12/04 Why is this copy here?
-    testBlock = cforLoop->testBlockGet()->copy(&map);
+    testBlock = cforLoop->testBlockGet();
```

Comment:
Copy is used, because expressions in the AST must have unique parent expressions.  Symbols may have multiple parents, but SymExprs (e.g.) have a unique parent.  This copy is not needed because testBlock is copied in the code that follows.  The copy here just wastes memory, time and ink.
Rather than remove the comment, I changed it into a note that a (moot) copy has been removed -- on the very remote chance that I'm wrong.  It's already been tested thoroughly, so my confidence level is out to five 9s.